standard/statements.md

```diff
@@ -352,6 +355,63 @@ The scope of a local constant is the block in which the declaration occurs. It i
 
 A local constant declaration that declares multiple constants is equivalent to multiple declarations of single constants with the same type.
 
+### §local-function-declarations-new-clause Local function declarations
+
+A *local_function_declaration* declares a local function.
+
+```ANTLR
+local_function_declaration
+    : local_function_header local_function_body
+    ;
+
+local_function_header
+    : local_function_modifiers? return_type identifier type_parameter_list?
+        ( formal_parameter_list? ) type_parameter_constraints_clause*
+    ;
+
+local_function_modifiers
+    : 'async'
+    | 'unsafe'
+    ;
+
+local_function_body
+    : block
+    | '=>' expression ';'
+    ;
+```
+
+Unless specified otherwise below, the semantics of all grammar elements is the same as for *method_declaration* ([15.6.1](classes.md#1561-general), read in the context of a local function instead of a method.
+
+A *local_function_declaration* may include one `async` ([§15.15](classes.md#1515-async-functions)) modifier and one `unsafe` ([§23.1](unsafe_code.md#231-general)) modifier. If the declaration includes the `async` modifier then the return type shall be `void` or a task type  ([§15.15.1](classes.md#15151-general)).
+
+A local function is declared at block scope, and that function may capture variables from the enclosing scope. Each call of the function requires captured variables to be definitely assigned before the call. The compiler shall determine which variables are definitely assigned on return.
+
+A local function may be called from a lexical point prior to its definition. However, it is a compile-time error for the function to be declared lexically prior to the declaration of a variable it wishes to capture.
+
+It is a compile-time error for a local function to declare a parameter or local variable with the same name as one declared in the enclosing scope.
+
+Local function declaration statements are always reachable.
```

Comment:
```C#
class C {
    int F() {
        if (true) {
            void G() {}
            return 1;
        } else {
            void H() {}
            return 2;
        }
    }
}
```

Is there a rule that says `return 1;` is reachable and `return 2;` is not? According to [13.3.2 Statement lists](https://github.com/dotnet/csharpstandard/blob/c86c9b0f464572b060dfa6688968237a4d23ad9e/standard/statements.md#1332-statement-lists), it depends on whether "The end point of the preceding statement is reachable." But if both local function declaration statements are reachable, then I don't see why the end point of `void G() {}` would be reachable and that of `void H() {}` would not.

Reply:
There isn't such a rule right now. The compiler determines that H() is reachable.